go/exocom/src/exocom/exocom.go

```diff
@@ -98,13 +105,26 @@ func (e *ExoCom) listenForErrors() {
 func (e *ExoCom) listenForMessages() {
 	for {
 		message := <-e.messageChannel
-		if e.routingManager.CanSend(message.Sender, message.Name) {
-			err := e.send(message)
-			if err != nil {
-				printLogError(e.logger.Error(err.Error()))
-			}
-		} else {
+		if !e.routingManager.CanSend(message.Sender, message.Name) {
 			printLogError(e.logger.Warning(fmt.Sprintf("Warning: Service '%s' is not allowed to broadcast the message '%s'", message.Sender, message.Name)))
+		} else {
+			if !e.hasSecurity {
+				err := e.send(message)
+				if err != nil {
+					printLogError(e.logger.Error(err.Error()))
+				}
+			} else {
+				securityResult := e.securityManager.ReceiveMessage(message)
+				if securityResult.MessageToSend != (structs.Message{}) {
```

Comment:
Thoughts on making MessageToSend a pointer to a message so we can compare against nil here?